git commit -m "feat: add Swagger UI documentation with SpringDoc"

MiguelAntonioRS opened the pull request.

MiguelAntonioRS committed:
git commit -m "feat: add Swagger UI documentation with SpringDoc"

MiguelAntonioRS commented:
feat: add Swagger UI documentation with SpringDoc

Adds interactive API documentation using SpringDoc OpenAPI. 
Now users can explore and test the API directly in the browser at `/swagger-ui/index.html`.

Changes:
- Added SpringDoc dependency
- Annotated controller methods with OpenAPI descriptions
- Minor config in application.properties

MiguelAntonioRS committed:
add standings endpoint to show league table

ricardozanini commented on pom.xml:
```diff
@@ -41,6 +41,12 @@
+		<dependency>
+			<groupId>org.springdoc</groupId>
+			<artifactId>springdoc-openapi-ui</artifactId>
+			<version>1.6.14</version>
```
Please move this version to the `properties` section.

MiguelAntonioRS replied:
I already did it when you can check it out

MiguelAntonioRS commented:
I hope I can continue helping you with this app.

ricardozanini commented:
Yes, please change the files as I requested there.

MiguelAntonioRS commented:
Hi!
Give time i´am finishing my engineering thesis
Thanks!

MiguelAntonioRS committed:
Moving to properties

MiguelAntonioRS committed:
Version

MiguelAntonioRS committed:
add support for multiple leagues and seasons

Adds support for loading and querying matches from different leagues and seasons. Example usage:

GET /api/matches?league=premier-league&season=2015

Includes:
- New CSV files for Premier League, La Liga, Bundesliga 2015 season
- Updated model with `league` and `season` fields
- Custom repository methods to filter by league/season
- Enhanced controller with optional query params

MiguelAntonioRS committed:
Merge branch 'master' of https://github.com/MiguelAntonioRS/soccer-stats

ricardozanini commented on pom.xml:
```diff
@@ -41,7 +46,16 @@
+		</dependency>
 		<dependency>
+=======
+>>>>>>> 4191ca7a67edefd0c1a44bdfc691a36d97c77917
```
Dirty from git rebase

ricardozanini commented on src/main/java/br/com/meetup/ansible/repository/MatchRepository.java:
```diff
@@ -15,5 +20,28 @@ public interface MatchRepository extends CrudRepository<Match, Integer> {
+
+	private List<Match> loadMatchesFromCsv(String filename, String league, String season) throws Exception {
+    List<Match> matches = new ArrayList<>();
+    Path path = Paths.get(ClassLoader.getSystemResource(filename).toURI());
```
Formatting

ricardozanini commented on src/main/java/br/com/meetup/ansible/service/MatchService.java:
```diff
@@ -12,4 +12,10 @@ public interface MatchService {
 	List<Match> listExactMatches(@NotEmpty String homeTeam, @NotEmpty String awayTeam);
 
+	public List<MatchDTO> findByLeagueAndSeason(String league, String season) {
+    return matchRepository.findByLeagueAndSeason(league, season)
```
Formatting

ricardozanini reviewed (changes requested):
Thank you! Please also remove the `.idea` folder from git.

MiguelAntonioRS commented:
Thank you for attending to my changes

MiguelAntonioRS committed:
Adds a simple JUnit 5 test for the MatchService class using Mockito.
Ensures that the service correctly returns all matches from the repository.
